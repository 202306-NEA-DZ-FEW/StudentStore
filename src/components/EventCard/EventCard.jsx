import React, { useState, useEffect } from "react";
import Image from "next/image";
import { FaCalendarAlt } from "react-icons/fa";
import Button from "../Buttons/Button";
import Link from "next/link";
const EventCard = () => {
    const data = [
        {
            id: 1,
            image: "/Events.png",
            title: "Book Buddies",
            description:
                "The buddies will read a book together and then spend time talking about the pictures, story and characters to help the story to come alive.üìö‚ú®",
            date: "Tuesday, 21 November",
            link: "https://www.eventbrite.com/e/book-buddies-tickets-704664549637?aff=ebdssbdestsearch&keep_tld=1",
        },
        {
            id: 2,
            image: "/Event3.jpg",
            title: " üåç How to be human in the time of the climate crisis",
            description:
                "Join us for a day's workshop to explore The Work that Reconnects. üåè Join students worldwide in shaping a sustainable future.",
            date: "Saturday, December 16 ",
            link: "https://www.eventbrite.com/e/how-to-be-human-in-the-time-of-the-climate-crisis-work-that-reconnects-tickets-747705686837?aff=ebdssbdestsearch",
        },
        {
            id: 3,
            image: "/Event2.jpg",
            title: "Discover Photography!",
            description:
                "Have you ever felt that you could be getting more from your digital camera? Would you like to unlock its full potential? If so, this half-day course is for you.",
            date: "Saturday, 16 December",
            link: "https://www.eventbrite.com/e/discover-photography-registration-36242436102?aff=ebdssbdestsearch&keep_tld=1",
        },
    ];
    const [event, setEvent] = useState(data);
    return (
        <section className='p-4'>
            <div className='mx-auto  lg:flex-row flex-col text-center  flex items-center overflow-hidden'>
                {event.map((event, eventIndex) => {
                    const { id, image, title, description, date, link } = event;
                    return (
                        <div key={id} className='p-4 '>
                            <article
                                className={` w-full  rounded-lg bg-white shadow-lg  `}
                            >
                                <header class='pt-1'>
                                    <h4 className=' font-bold lg:text-lg text-[#585785] '>
                                        {title}
                                    </h4>
                                </header>
                                <div class=' space-y-4 p-4 mb-2'>
                                    <Image
                                        src={image}
                                        alt={title}
                                        width={275}
                                        height={250}
                                        className='object-cover  mx-auto block  rounded-md '
                                    />
                                    <p className='max-w-[15rem] sm:max-w-[25rem] md:max-w-[30rem] lg:max-w-[35rem] xl:max-w-[40rem] mx-auto text-[#585785] leading-loose'>
                                        {description}
                                    </p>
                                </div>
                                <footer class='flex lg:justify-around  lg:flex-row flex-col p-2 '>
                                    <div className='flex space-x-2 justify-center text-center text-[#585785] '>
                                        {" "}
                                        <FaCalendarAlt
                                            size={25}
                                        ></FaCalendarAlt>{" "}
                                        <p className=' text-center text-lg font-semibold'>
                                            {date}
                                        </p>
                                    </div>
                                    <Button className='bg-[#585785] py-1 lg:px-2 border-none px-1  w-24 m-auto  '>
                                        {" "}
                                        <Link href={`${link}`}>
                                            {" "}
                                            Join us{" "}
                                        </Link>{" "}
                                    </Button>
                                </footer>
                            </article>
                        </div>
                    );
                })}
            </div>
        </section>
    );
};

export default EventCard;
