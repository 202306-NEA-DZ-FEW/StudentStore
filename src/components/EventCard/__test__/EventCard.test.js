import renderer from "react-test-renderer";
import EventCard from "../EventCard";

it("renders correctly", () => {
    const tree = renderer.create(<EventCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
