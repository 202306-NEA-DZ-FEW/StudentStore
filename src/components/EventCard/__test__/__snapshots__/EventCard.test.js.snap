// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="p-4"
>
  <div
    className="mx-auto  lg:flex-row flex-col text-center  flex items-center overflow-hidden"
  >
    <div
      className="p-4 w-120 h-120  flex-1"
    >
      <article
        className=" w-full   rounded-lg bg-white shadow-lg  "
      >
        <header
          class="pt-1 flex-shrink"
        >
          <h4
            className=" font-bold md:text-lg text-[#585785] "
          >
            Book Buddies
          </h4>
        </header>
        <div
          class=" space-y-4 md:w-[60vw]   lg:w-[30vw] p-4 mb-2"
        >
          <img
            alt="Book Buddies"
            className="object-cover  mx-auto block  rounded-md "
            data-nimg="1"
            decoding="async"
            height={250}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fbookbuddies.jpg&w=640&q=75"
            srcSet="/_next/image?url=%2Fbookbuddies.jpg&w=384&q=75 1x, /_next/image?url=%2Fbookbuddies.jpg&w=640&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={275}
          />
          <p
            className="  text-[#585785] "
          >
            The buddies will read a book together and then spend time talking about the pictures, story and characters to help the story to come alive.üìö‚ú®
          </p>
        </div>
        <footer
          class="flex lg:justify-around items-center lg:flex-row flex-col p-2 "
        >
          <div
            className="flex items-center space-x-2 justify-center text-center text-[#585785] "
          >
             
            <svg
              className=" flex items-center"
              fill="currentColor"
              height={25}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width={25}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              />
            </svg>
             
            <p
              className=" text-center text-lg font-semibold"
            >
              Tuesday, 21 November
            </p>
          </div>
          <button
            className="border-2 border-[#FF8A57] rounded-lg text-white shadow-sm text-md hover:bg-white hover:text-[#FF8A57] hover:border-2 hover:border-[#FF8A57] hover:duration-300 font-bold bg-[#585785] py-1 lg:px-2 border-none px-1 w-24 m-auto"
          >
             
            <a
              href="https://www.eventbrite.com/e/book-buddies-tickets-704664549637?aff=ebdssbdestsearch&keep_tld=1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              target="_blank"
            >
               
              Join us
               
            </a>
             
          </button>
        </footer>
      </article>
    </div>
    <div
      className="p-4 w-120 h-120  flex-1"
    >
      <article
        className=" w-full   rounded-lg bg-white shadow-lg  "
      >
        <header
          class="pt-1 flex-shrink"
        >
          <h4
            className=" font-bold md:text-lg text-[#585785] "
          >
            the climate crisis
          </h4>
        </header>
        <div
          class=" space-y-4 md:w-[60vw]   lg:w-[30vw] p-4 mb-2"
        >
          <img
            alt="the climate crisis"
            className="object-cover  mx-auto block  rounded-md "
            data-nimg="1"
            decoding="async"
            height={250}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2FEvent.jpg&w=640&q=75"
            srcSet="/_next/image?url=%2FEvent.jpg&w=384&q=75 1x, /_next/image?url=%2FEvent.jpg&w=640&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={275}
          />
          <p
            className="  text-[#585785] "
          >
            Join us for a day's workshop to explore The Work that Reconnects. üåè Join students worldwide in shaping a sustainable future.
          </p>
        </div>
        <footer
          class="flex lg:justify-around items-center lg:flex-row flex-col p-2 "
        >
          <div
            className="flex items-center space-x-2 justify-center text-center text-[#585785] "
          >
             
            <svg
              className=" flex items-center"
              fill="currentColor"
              height={25}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width={25}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              />
            </svg>
             
            <p
              className=" text-center text-lg font-semibold"
            >
              Saturday, 16 December  
            </p>
          </div>
          <button
            className="border-2 border-[#FF8A57] rounded-lg text-white shadow-sm text-md hover:bg-white hover:text-[#FF8A57] hover:border-2 hover:border-[#FF8A57] hover:duration-300 font-bold bg-[#585785] py-1 lg:px-2 border-none px-1 w-24 m-auto"
          >
             
            <a
              href="https://www.eventbrite.com/e/how-to-be-human-in-the-time-of-the-climate-crisis-work-that-reconnects-tickets-747705686837?aff=ebdssbdestsearch"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              target="_blank"
            >
               
              Join us
               
            </a>
             
          </button>
        </footer>
      </article>
    </div>
    <div
      className="p-4 w-120 h-120  flex-1"
    >
      <article
        className=" w-full   rounded-lg bg-white shadow-lg  "
      >
        <header
          class="pt-1 flex-shrink"
        >
          <h4
            className=" font-bold md:text-lg text-[#585785] "
          >
            Discover Photography!
          </h4>
        </header>
        <div
          class=" space-y-4 md:w-[60vw]   lg:w-[30vw] p-4 mb-2"
        >
          <img
            alt="Discover Photography!"
            className="object-cover  mx-auto block  rounded-md "
            data-nimg="1"
            decoding="async"
            height={250}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fphotography.jpg&w=640&q=75"
            srcSet="/_next/image?url=%2Fphotography.jpg&w=384&q=75 1x, /_next/image?url=%2Fphotography.jpg&w=640&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={275}
          />
          <p
            className="  text-[#585785] "
          >
            Have you ever felt that you could be getting more from your digital camera? Would you like to unlock its full potential? 
          </p>
        </div>
        <footer
          class="flex lg:justify-around items-center lg:flex-row flex-col p-2 "
        >
          <div
            className="flex items-center space-x-2 justify-center text-center text-[#585785] "
          >
             
            <svg
              className=" flex items-center"
              fill="currentColor"
              height={25}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width={25}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              />
            </svg>
             
            <p
              className=" text-center text-lg font-semibold"
            >
              Saturday, 16 December
            </p>
          </div>
          <button
            className="border-2 border-[#FF8A57] rounded-lg text-white shadow-sm text-md hover:bg-white hover:text-[#FF8A57] hover:border-2 hover:border-[#FF8A57] hover:duration-300 font-bold bg-[#585785] py-1 lg:px-2 border-none px-1 w-24 m-auto"
          >
             
            <a
              href="https://www.eventbrite.com/e/discover-photography-registration-36242436102?aff=ebdssbdestsearch&keep_tld=1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              target="_blank"
            >
               
              Join us
               
            </a>
             
          </button>
        </footer>
      </article>
    </div>
  </div>
</section>
`;
